backend/application/syson-application-configuration/src/main/java/org/eclipse/syson/application/migration/SuccessionAsUsageMigrationParticipant.java

```diff
@@ -0,0 +1,117 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.syson.application.migration;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.eclipse.emf.ecore.EClass;
+import org.eclipse.emf.ecore.EClassifier;
+import org.eclipse.sirius.components.emf.migration.api.IMigrationParticipant;
+import org.eclipse.sirius.emfjson.resource.JsonResource;
+import org.eclipse.syson.sysml.SuccessionAsUsage;
+import org.eclipse.syson.sysml.SysmlPackage;
+import org.springframework.stereotype.Service;
+
+/**
+ * Migration participant used to migrate old SysML mode previous to 2025.4.0.
+ *
+ * <p>
+ * It is intended to overcome the previous implementation that serialized data in <i>source</i> and <i>target</i>
+ * features of {@link SuccessionAsUsage} whereas now those features are derived and non editable. See detail at
+ * https://github.com/eclipse-syson/syson/issues/1030
+ * </p>
+ *
+ * @author Arthur Daussy
+ */
+@Service
+public class SuccessionAsUsageMigrationParticipant implements IMigrationParticipant {
```

Comment:
Could you please add an integration test for this migration participant?

Reply:
Done.